src/transport.js

```diff
@@ -65,21 +65,44 @@ function generateTransport(winston, winstonTransport) {
      * @returns {void}
     */
     log(info) {
-      //  If we have specified to log in JSON format, then stringify
-      //
-      //  Also, if the `info` object which is passed from winston has more
-      //  that 2 default keys (`message` and `level`), then that means that
-      //  the `log` function was called with more two than keys
-      //  (referred to as metadata), so in that scenario we stringify the
-      //  entire message since it is not obvious what format the user wants.
+      //  This function is required since Winston passes in metadata in a strange way.
+      //  Rather than giving us a complete object containing all metadata, it hides it away
+      //  in the splat attribute and passes in only the first piece of metadata
       //  
-      //  log function, without being given a log (like below) will just log
-      //  the level, which in this case is our message.
-      if (this.json || Object.keys(info).length > 2) {
-        this.logger.log(stringify(info));
+      //  This seems to be an existing issue faced by devs:
+      //  https://stackoverflow.com/a/60866937
+      const returnMetadata = (meta) => {
+        // You can format the splat yourself
+        const splat = meta[Symbol.for('splat')];
+
+        if (splat && splat.length) {
+          return splat.reduce((result, current) => Object.assign(result, current), {});
+        }
+
+        return {};
+      };
+
+      const metadata = returnMetadata(info);
+
+      if (this.json) {
+        //  If we are to output JSON, we create the full object containing level, message and metadata.
+        //
+        //  We don't serialize here but pass in an object since InsightLogger appends extra keys based on
+        //  configuration, e.g. timestamp before serializing to JSON
+        //
+        //  We also don't specify the first argument of `level` since the winston `info` contains it.
+        //  If we did provide it InsightLogger would append an extra redundant `_level` key.
+        this.logger.log({
+          ...info,
+          ...metadata,
+        });
+      } else if (Object.keys(info).length > 2) {
+        //  If we are not outputting to JSON and have metadata, we use the same format as Winston
+        const message = `${info.level}: ${info.message} ${stringify(metadata)}`;
```

Comment:
isn't this different from the current behaviour, where we would still log as JSON?

Reply:
yup.

although we shouldn't, we haven't selected to log using JSON, so therefore we should log as string.
once again mentioning that this is the same functionality as winston with `format.simple()`